File: Printpress.Application/Inventory/DTO/PurchaseInvoiceCreateDto.cs
namespace Printpress.Application;

public class PurchaseInvoiceCreateDto
{
    public DateTime InvoiceDate { get; set; }
    public string SupplierName { get; set; }
    public string AttachmentFilePath { get; set; }
    public decimal? PaidNow { get; set; }
    public bool? ReceiveNow { get; set; }
    public List<PurchaseInvoiceLineCreateDto> Lines { get; set; }
}

File: Printpress.UI/Printpress_Angular/src/app/features/inventory/models/purchase-invoice-create.dto.ts
export interface PurchaseInvoiceLineCreateDto {
  inventoryItemId: string;
  quantity: number;
  unitPrice: number;
}

export interface PurchaseInvoiceCreateDto {
  invoiceNumber: string;
  invoiceDate: string;
  supplierName: string;
  attachmentFilePath: string;
  paidNow: number;
  receiveNow: boolean;
  lines: PurchaseInvoiceLineCreateDto[];
}
